### PR TITLE
Add reporting of console log checking stage

### PR DESCRIPTION
**Pull request by veruu**
matches changes done in https://github.com/RH-FMK/skt/pull/325

### CODE REVIEW

File: sktm/reporter.py [hunk 1]
```diff
@@ -28,6 +28,7 @@
 SUMMARY_MERGE_FAILURE = 1
 SUMMARY_BUILD_FAILURE = 2
 SUMMARY_TEST_FAILURE = 3
+SUMMARY_CALL_TRACE = 4
```
Comment (spbnick) [inline sktm/reporter.py]: This name is somewhat confusing and doesn't match the pattern so far. Could you call it something like `SUMMARY_CALL_TRACE_FOUND`, or `SUMMARY_TRACE_FOUND`, or `SUMMARY_TRACE_WARNING`?

File: sktm/reporter.py [hunk 2]
```diff
@@ -246,7 +248,27 @@ def __create_data(self, merge_report, result_set_list=[]):
                                                       test_result_dir,
                                                       index)
 
-            full_report += '\n' + build_report + run_report
+            console_result = next((test_result for test_result in test_run
+                                   if test_result.endswith(
+                                           'console_check.result'
+                                   )), None)
```
Comment (spbnick) [inline sktm/reporter.py]: This seems really neat, but could you please add a brief comment explaining what you're doing here and in cases above? Or perhaps rewrite those so they're more obvious? Thanks :)

Comment (veruu) [reply sktm/reporter.py]: I'll add a separate commit with explanation

**Comment by veruu** [comment]
@spbnick changed the constant name and added explanation for the `next()` calls

**Comment by spbnick** [comment]
Thank you, this looks good, but I'd like to ask for more explanations :)

**Comment by veruu** [comment]
@spbnick can you elaborate on what other information would you like to see in the explanation? I explained *why* the `next()` call is used instead of other things and added a link which has more details. Everything about the syntax itself is in the docs which is not something that IMHO should be copied over.

**Comment by spbnick** [comment]
Sure!

* [ ] I would like to have a more explicit description of the `result_set_list` argument in the `__create_data()` docstring. Specifically, what the sets contain exactly.
* [ ] In addition to the explanation of the technique, put a single-line comment above each of the `*_result = next(...)` blocks, saying something like "Get first blah result filename, if any". E.g.:

            # Get first build result filename, if any
            build_result = next((test_result for test_result in test_run
                                 if test_result.endswith('build.result')),
                                None)

Do you think that would make sense?

**Comment by veruu** [comment]
I'll add another commit for the explanation of the first point (since it's not really related to the original commit this pull came with). The second one seems redundant since the explanation is only a few lines above the line. I'd actually like to extract the similar parts into a method and parametrize what should be extracted/returned but I didn't have time to do it yet, and then the explanation can be mentioned in the docstring of that method as well, what do you think about that solution?

**Comment by veruu** [comment]
Haha, that happens! Thanks for checking everything :)

**Pull request merged**